msu/hooks/items/item_container.nut

```diff
@@ -94,4 +94,35 @@
 
 		return unequip(_item);
 	}
+
+	q.getItemsByFunction <- function( _function )
+	{
+		local ret = [];
+
+		foreach (slot, spaces in ::Const.ItemSlotSpaces)
+		{
+			for (local i = 0; i < spaces; i++)
```

Comment:
This should probably just use `getItemsByFunctionAtSlot`:
`ret.extend(getItemsByFunctionAtSlot(slot, _function);`

Reply:
That results in way more iterations and array declarations than the current implementation. I don't like that.

Reply:
Is it worse than resusing the same code in two functions with the only difference being another loop though

Reply:
nvm